### PR TITLE
move zabbixapi gem installation into own class

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: manifests/zabbixapi.pp [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      fail("Zabbix ${zabbix_version} is not supported!")
+    }
+  }
+
+  $compile_packages = $facts['os']['family'] ? {
+    'RedHat' => ['make', 'gcc-c++', 'rubygems', 'ruby'],
+    'Debian' => ['make', 'g++', 'ruby-dev', 'ruby', 'pkg-config',],
+    default  => [],
+  }
+  ensure_packages($compile_packages)
```
Comment (bastelfreak) [inline manifests/zabbixapi.pp:24]: that is deprecated, please use the newer version:
```suggestion
  stdlib::ensure_packages($compile_packages)
```

Comment (evgeni) [reply manifests/zabbixapi.pp:24]: can I use this now w/o bumping stdlib to >= 9?

Comment (ekohl) [reply manifests/zabbixapi.pp:24]: I don't think so

File: manifests/zabbixapi.pp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# @param zabbix_version This is the zabbix version.
+# @param puppetgem Provider for the zabbixapi gem package.
+class zabbix::zabbixapi (
+  $zabbix_version = $zabbix::params::zabbix_version,
+  $puppetgem = $zabbix::params::puppetgem,
+) inherits zabbix::params {
+  # Determine correct zabbixapi version.
+  case $zabbix_version {
+    /^[56]\.[024]/: {
+      $zabbixapi_version = '5.0.0-alpha1'
```
Comment (ekohl) [inline manifests/zabbixapi.pp:12]: Note to self: https://rubygems.org/gems/zabbixapi does only list a 5.0.0-alpha1 from 2021.

File: manifests/zabbixapi.pp [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# @summary This will install the zabbixapi gem.
+#
+# @param zabbix_version This is the zabbix version.
+# @param puppetgem Provider for the zabbixapi gem package.
+class zabbix::zabbixapi (
+  $zabbix_version = $zabbix::params::zabbix_version,
+  $puppetgem = $zabbix::params::puppetgem,
```
Comment (ekohl) [inline manifests/zabbixapi.pp:7]: Should this be `String[1]` for both?

Comment (evgeni) [reply manifests/zabbixapi.pp:7]: In theory, yes.
In reality, we do not enforce these anywhere else, so I find it odd to do it here.

(read: we should do it globally, but not in this PR)

**Pull request merged**